src/node/internal/internal_fs_sync.ts

```diff
@@ -0,0 +1,1066 @@
+  fd = getValidatedFd(fd);
+  parseFileMode(mode, 'mode');
+  if (cffs.stat(fd, { followSymlinks: true }) == null) {
+    throw new Error('bad file descriptor'); // not found or not accessible.
+  }
+  // We do not implement chmod in any meaningful way as our filesystem
+  // has no concept of user-defined permissions. Once we validate the inputs
+  // we just return as a non-op.
+  // The reason we call cffs.stat is to ensure, at the very least, that
+  // the fd is valid and would otherwise be accessible.
+}
+
+export function fchownSync(fd: number, uid: number, gid: number): void {
+  fd = getValidatedFd(fd);
+  validateInteger(uid, 'uid', -1, kMaxUserId);
+  validateInteger(gid, 'gid', -1, kMaxUserId);
+  if (cffs.stat(fd, { followSymlinks: true }) == null) {
+    throw new Error('bad file descriptor'); // not found or not accessible.
+  }
+  // We do not implement chown in any meaningful way as our filesystem
+  // has no concept of ownership. Once we validate the inputs we just
+  // return as a non-op.
+  // The reason we call accessSync is to ensure, at the very least, that
+  // the path exists and would otherwise be accessible.
+}
+
+export function fdatasyncSync(fd: number): void {
+  fd = getValidatedFd(fd);
+  // We do not implement fdatasync in any meaningful way. At most we
+  // will validate that the fd is valid and would otherwise be accessible.
+  if (cffs.stat(fd, { followSymlinks: true }) == null) {
+    throw new Error('bad file descriptor'); // not found or not accessible.
+  }
+}
+
+export function fstatSync(
+  fd: number,
+  options: { bigint: boolean } = { bigint: false }
+): Stats {
+  fd = getValidatedFd(fd);
+  validateObject(options, 'options');
+  const { bigint } = options;
+  validateBoolean(bigint, 'options.bigint');
+  const stat = cffs.stat(fd, { followSymlinks: true });
+  if (stat == null) {
+    throw new Error('bad file descriptor'); // not found or not accessible.
+  }
+  return new Stats(kBadge, stat, { bigint });
+}
+
+export function fsyncSync(fd: number): void {
+  fd = getValidatedFd(fd);
+  // We do not implement fdatasync in any meaningful way. At most we
+  // will validate that the fd is valid and would otherwise be accessible.
+  if (cffs.stat(fd, { followSymlinks: true }) == null) {
+    throw new Error('bad file descriptor'); // not found or not accessible.
+  }
+}
+
+export function ftruncateSync(fd: number, len: number = 0): void {
+  validateUint32(len, 'len');
+  cffs.truncate(getValidatedFd(fd), len);
+}
+
+function getDate(time: string | number | bigint | Date): Date {
+  if (typeof time === 'number') {
+    return new Date(time);
+  } else if (typeof time === 'bigint') {
+    return new Date(Number(time));
+  } else if (typeof time === 'string') {
+    return new Date(time);
+  } else if (time instanceof Date) {
+    return time;
+  }
+  throw new ERR_INVALID_ARG_TYPE(
+    'time',
+    ['string', 'number', 'bigint', 'Date'],
+    time
+  );
+}
+
+export function futimesSync(
```

Comment:
What about `glob` / `globSync`? At the very least could be worth having a comment / placeholder.